store/storetest/channel_store.go

```diff
@@ -2714,7 +2761,8 @@ func testChannelStoreGetMembersByIds(t *testing.T, ss store.Store) {
 	require.Nil(t, err)
 
 	m1 := &model.ChannelMember{ChannelId: o1.Id, UserId: model.NewId(), NotifyProps: model.GetDefaultChannelNotifyProps()}
-	store.Must(ss.Channel().SaveMember(m1))
+	_, err = ss.Channel().SaveMember(m1)
+	require.Nil(t, err)
 
```

Comment:
a way to fix the shadowing here is by doing:
```suggestion
var members *model.ChannelMembers
if members, err = ss.Channel().GetMembersByIds(m1.ChannelId, []string{m1.UserId}); err != nil {
```

Reply:
@Hyaxia fyi. You'll have to make the above change to this line as well as update line `2767` to fix `err` shadowing.

Reply:
Yeah I thought about fixing it that way, but I the explanation for why it happens still dosent satisfy me....
Commented below.

> > > > @streamer45
> > > > > Line 2819 defines a new scope (if block) and you are essentially creating a new variable `err` inside it which shadows the original `err` var.
> > > > 
> > > > 
> > > > But the thing is that the variable `members` is new, so I have to use the `:=` operator even though the `err` variable is already declared, and the compiler should know how to handle it, and he does handle it in many places, except for this one, for some reason that I cannot explain.
> > > 
> > > 
> > > You get the warning in this case because the original `err` var is used again in the outer scope after that if/else block.
> > 
> > 
> > @streamer45 Sorry I dont get it...
> > The variable member is first declared at line 2767, and it says that the variable `err` on the same line shadows another variable named `err`.
> > But the thing is I HAVE to use the `:=` operator to create the new `memebers` variable...
> > The error says its about shadowing a variable on an earlier line, so I dont see a connection to the the lines after the if...else clause.
> 
> The problem is that `err` already exists in the outer scope and you are using `:=` in an inner scope (if/else block) so you are effectively declaring two new variables (`members` and `err`) inside the if/else scope. Take a look at https://golang.org/doc/effective_go.html#redeclaration for reference.

In the url you have provided, it is said
```
which looks as if it declares d and err. Notice, though, that err appears in both statements. This duplication is legal: err is declared by the first statement, but only re-assigned in the second.
.
.
.
there is at least one other variable in the declaration that is being declared anew.
```

and that is exactly as I said, I have a new variable, `members` which is declared in that if statement, and the already declared variable, `err` which is just assigned a new value.

For example, how is this example different from - 
```
a := 3
	if b, a := 2, 4; b != 3 {
		
	}
```

@ali-farooq0 I saw the way you suggest to solve it, and I have also thought about it as a solution, but I still really want to understand why this case it acts differently from what I thought it would.
Thanks!